omigami/spectra_matching/spectra_matching.py

```diff
@@ -183,41 +187,70 @@ def _make_batch_requests(
         parameters: SpectraMatchingParameters,
         endpoint: str,
     ) -> List[pd.DataFrame]:
+        self._failed_spectra = []
         batch = []
-        requests = []
+        results = []
 
         for spectrum in spectra_generator:
+            if self._get_cache_id(spectrum) in self._cached_results:
+                results.append(self._cached_results[self._get_cache_id(spectrum)])
+                continue
+
```

Comment:
I have more readable suggestion:
```suggestion
            cache_id = self._get_cache_id(spectrum)
            if  cache_id in self._cached_results:
                results.append(self._cached_results[cache_id])
                continue

```